### ISSUE TITLE
Error Building "react-native build-android"

### ISSUE THREAD

**Issue by 2acoin** [turtlecoin/turtlecoin-mobile-wallet#80]
We pulled down a copy of the current version of **turtlecoin-mobile-wallet** and can not get it to compile. We have installed all the pre-requisite libraries and then we execute the following;

`react-native run-android`
 
It begins building and gets to 98% and then throws the following error;

```
C:\Users\user1\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\34928662a7031e77600c9f37b8bf183c\res\values\values.xml:251:5-69: AAPT: err
or: resource android:attr/fontVariationSettings not found.

C:\Users\user1\.gradle\caches\transforms-1\files-1.1\appcompat-v7-27.1.1.aar\34928662a7031e77600c9f37b8bf183c\res\values\values.xml:251:5-69: AAPT: err
or: resource android:attr/ttcIndex not found.

error: failed linking references.

```

The strange thing about this, a few days ago this was working fine. Any ideas?

Googling for answers talks about changing the `compileSDKVersion`  which we have tried without any success. We have also tried changing other support libraries but have thus far been unsuccessful.

**Comment by zpalmtree** [comment]
Not sure I'm afraid - I have never encountered this. As you said, it worked before - and no code has been changed, so must be something on your system.

I was working on upgrading the codebase to react 59, just never got around to pushing it up. That could potentially help. I'll try and push that up to a branch in a bit.

I assume you tried a fresh clone? For some reason, sometimes the build inexplicably fails, and works upon retrying.

**Comment by 2acoin** [comment]
Yes, we have experienced those "odd" errors and rebuild then works... However, this latest error is a pain as it just will not go away no matter what we try.

**Comment by 2acoin** [comment]
Is there plans for work to continue on this wallet?

**Comment by zpalmtree** [comment]
Not many, it works and does enough, I'm planning a rewrite in Qt+C++ at *some point*, not sure when.

I just pushed the react 59 updates to https://github.com/turtlecoin/turtlecoin-mobile-wallet/tree/react-native-59 , see if that works for you

**Comment by 2acoin** [comment]
Thanks, downloaded the latest react 59 version and we'll give it a try. We setup a clean development environment for this build so keeping our fingers crossed.

**Comment by 2acoin** [comment]
@zpalmtree Have you been successful in building using the `react 59 version` ?

We created a fresh build environment with all the bells and whistles needed to make the build. We then ran the build and we get this error;

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':app:preDebugBuild'.
> Could not resolve all task dependencies for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve project :@react-native-community_netinfo.
     Required by:
         project :app
      > Unable to find a matching configuration of project :@react-native-community_netinfo: None of the consumable configurations have attributes.
   > Could not resolve project :@react-native-community_async-storage.
     Required by:
         project :app
      > Unable to find a matching configuration of project :@react-native-community_async-storage: None of the consumable configurations have attribues.

```
Just wondering if this has been successfully built before we spend a lot of time trying to get it to work.

**Comment by zpalmtree** [comment]
It worked when I made the changes, but appears to have broken now due to libraries migrating to androidX, I guess my dependencies weren't pinned well enough. Working on a fix.

**Comment by zpalmtree** [comment]
The original cause of the issue appears to have been this androidX upgrade, where google updated their libraries yesterday or the day before. I have fixed the issues and it builds fine with a fresh clone.

```
git clone https://github.com/turtlecoin/turtlecoin-mobile-wallet.git
git checkout react-native-59
yarn install
yarn start
```

Make sure the jetify process runs (will take a *long* time). Should be done automatically upon `yarn install`.

**Comment by 2acoin** [comment]
Awesome... We will give it a try, wiping all traces of previous version as we were experimenting. I suppose the change in the AndroidX upgrade is what impacted us. As that is about the time frame we saw everything turn to crap...

**Comment by 2acoin** [comment]
Dumb question - 
`yarn start` - appears to build the bundle just fine. However, we can not install a bundle to the device for testing. So we tried to execute the following;
`yarn build-android` - which failed with this error;

```
> Task :@react-native-community_netinfo:compileReleaseJavaWithJavac FAILED
D:\MySource\Android\turtle-wallet\node_modules\@react-native-community\netinfo\android\src\main\java\com\reactnativecommunity\netinfo\ConnectivityRece
iver.java:12: error: package android.support.v4.net does not exist
import android.support.v4.net.ConnectivityManagerCompat;
                             ^
D:\MySource\Android\turtle-wallet\node_modules\@react-native-community\netinfo\android\src\main\java\com\reactnativecommunity\netinfo\ConnectivityRece
iver.java:154: error: cannot find symbol
                    ConnectivityManagerCompat.isActiveNetworkMetered(getConnectivityManager());
                    ^
  symbol:   variable ConnectivityManagerCompat
  location: class ConnectivityReceiver
2 errors

FAILURE: Build failed with an exception.

```
Any ideas?

**Comment by zpalmtree** [comment]
Err, that's the same error that occurs when jetify doesn't run. Not sure what needs to be modified to get the release files to be 'jetified'... I'll look into it tonight, probably.

**Comment by zpalmtree** [comment]
Just tested now, I used `yarn deploy-android`, but if that works then `yarn build-android` should also work. I did not get the error you did - only an error about duplicate resources, which is now fixed.

Maybe try manually running `./node_modules/bin/jetify`?

**Comment by 2acoin** [comment]
We're running our build on a Windows machine. The build scripts do not work out of the box so we have to modify the steps a bit. Basically, this is what we have done;

```
git clone https://github.com/turtlecoin/turtlecoin-mobile-wallet.git
git checkout react-native-59
cd turtlecoin-mobile-wallet
yarn install
gradlew.bat assembleRelease

```
It runs thru the build and then throws the error we posted above.

We found a discussion about the AndroidX issue with the `react-native `components. There are a few suggested approaches that we will try from the post.

[https://github.com/react-native-community/discussions-and-proposals/issues/129](https://github.com/react-native-community/discussions-and-proposals/issues/129)
